modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala

```diff
@@ -6,24 +6,19 @@ import java.nio.file.Path
 import scala.language.higherKinds
 import scala.reflect.ClassTag
 
-import cats.data.NonEmptyList
-import cats.effect.Sync
+import cats.effect.{ Blocker, ContextShift, Sync }
 import cats.implicits._
-import com.typesafe.config.{ ConfigRenderOptions, Config => TypesafeConfig }
+import com.typesafe.config.ConfigRenderOptions
 import pureconfig._
 import pureconfig.error.ConfigReaderException
 
 package object catseffect {
 
-  @deprecated("Root will be treated as the default namespace", "0.12.0")
-  val defaultNameSpace = ""
+  private[catseffect] type ConfReader[A] = Derivation[ConfigReader[A]]
```

Comment:
We usually don't add a type alias `Derivation[ConfigReader[A]]` on a per-file basis, as it may make the code less readable (e.g. `ConfReader` is hard to distinguish from `ConfigReader`). I am not against creating an alias for this since it's used so heavily in our code base, but if we do it we'll end up doing that on a project-wide basis.

Note that you can still use the nice context bound syntax for most other type classes, you just need to keep the reader as an "explicit" implicit param.

Reply:
Mixing context bounds and implicits are a bad practice. But I really didn't liked the named implicits since neither of them is directly used. However, I understand the problem with the typealias, so if you prefer I can revert to named implicits and  if someday you guys decide to add the alias on the core module, then we could use context bounds.

Reply:
I understand it's a bad practice and I totally understand the preference for context bounds instead of named implicits. However, for the sake of keeping the PR focused on a single thing and avoiding a confusing type alias let's leave this as it was, if that's ok with you :)